src/modules/Api.ts

```diff
@@ -183,6 +187,96 @@ export class Api {
     });
   }
 
+  public async canonicalOrder({
+    isBuy,
+    isDecreaseOnly,
+    baseMarket,
+    quoteMarket,
+    amount,
+    limitPrice,
+    signedTriggerPrice,
+    limitFee,
+    makerAccountNumber,
+    makerAccountOwner,
+    expiration,
+    fillOrKill,
+    postOnly,
+    triggerPrice,
+    marginDeposit,
+    payoutAmount,
+    clientId,
+    setExpirationOnFill,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
```

Comment:
we should break this up into two fields: the order itself, and some other options like fillOrKill, setExpirationOnFill, etc

we should also have default options for the vast majority of these

we may not even want to expose some things here like `triggerPrice/signedTriggerPrice/isDecreaseOnly`
let's optimize for typical marketmakers

Another thing is that we might want to do is use the `ApiMarketName` enum in `types.ts` instead of having the market-makers specify the baseMarket and quoteMarket themselves

Reply:
I'm not seeing a default for makerAccountOwner but I found a default for expiration & all of the non order fields are already optional